src/app/components/wrapper-components/online-game-wrapper/online-game-wrapper.component.ts

```diff
@@ -602,73 +624,99 @@ export class OnlineGameWrapperComponent extends GameWrapper implements OnInit, O
+        const lastIndex: number = this.getLastIndex();
+        const user: Player = this.getPlayer();
         this.stopCountdownsFor(Player.of(player));
         const opponent: IUserId = Utils.getNonNullable(this.opponent);
         if (player === this.observerRole) {
             // the player has run out of time, he'll notify his own defeat by time
-            this.notifyTimeoutVictory(Utils.getNonNullable(opponent.doc.username), this.getPlayerName());
+            this.notifyTimeoutVictory(Utils.getNonNullable(opponent.doc.username),
+                                      user,
+                                      lastIndex,
+                                      this.getPlayerName());
         } else {
             if (this.endGame) {
                 display(true, 'time might be better handled in the future');
             } else if (this.opponentIsOffline()) { // the other player has timed out
-                this.notifyTimeoutVictory(this.getPlayerName(), Utils.getNonNullable(opponent.doc.username));
+                this.notifyTimeoutVictory(this.getPlayerName(),
+                                          user,
+                                          player,
+                                          Utils.getNonNullable(opponent.doc.username));
                 this.endGame = true;
             }
         }
     }
     public acceptRematch(): boolean {
-        if (this.isPlaying() === false) {
+        if (this.isPlaying() === false) { // TODOTODO eeeeeh
             return false;
         }
         const currentPartId: ICurrentPartId = {
             id: this.currentPartId,
             doc: this.currentPart.doc,
         };
-        this.gameService.acceptRematch(currentPartId);
+        const user: Player = this.getPlayer();
+        const lastIndex: number = this.getLastIndex();
+        this.gameService.acceptRematch(currentPartId, user, lastIndex);
         return true;
     }
     public proposeRematch(): boolean {
-        if (this.isPlaying() === false) {
+        if (this.isPlaying() === false) { // TODOTODO eeeeh
             return false;
         }
-        this.gameService.proposeRematch(this.currentPartId, this.getPlayer());
+        const user: Player = this.getPlayer();
+        const lastIndex: number = this.getLastIndex();
+        this.gameService.proposeRematch(this.currentPartId, lastIndex, user);
         return true;
     }
     public proposeDraw(): void {
-        this.gameService.proposeDraw(this.currentPartId, this.getPlayer());
+        const user: Player = this.getPlayer();
+        const lastIndex: number = this.getLastIndex();
+        this.gameService.proposeDraw(this.currentPartId, lastIndex, user);
     }
     public acceptDraw(): void {
-        this.gameService.acceptDraw(this.currentPartId);
+        const lastIndex: number = this.getLastIndex();
+        const user: Player = this.getPlayer();
+        this.gameService.acceptDraw(this.currentPartId, lastIndex, user);
```

Comment:
c'est voulu l'alternance user/lastIndex lastIndex/user ? :D
Sinon ici j'serais pas mécontent avec une seule ligne: `this.gameService.acceptDraw(currentPartId, this.getLastIndex(), this.getPlayer())`

Reply:
Je trouvais que 80+80+80 c'est plus joli que 100, mais ça va, 100 c'est ok.
Sinon oui, j'vais uniformiser lastIndex->user